lnrpc/routerrpc/router.proto

```diff
@@ -125,8 +124,27 @@ message PaymentStatus {
     The taken route when state is SUCCEEDED.
     */
     lnrpc.Route route = 3;
+
+    /**
+    List of all shard statuses to this payment_hash [EXPERIMENTAL].
+    */
+    repeated ShardStatus shards = 4 [json_name = "shards"];
```

Comment:
I wouldn't mind leaving reporting of shards out of scope initially and just make sure that the payment-level reporting is mpp aware.